lib/OpenQA/Jobs/Constants.pm

```diff
@@ -78,3 +78,11 @@ our @EXPORT = qw(
   MODULE_RESULTS
   TIMEOUT_EXCEEDED
 );
+
+# define mapping from any specific job result to a "generalized" form where the different incomplete results are not distinguished
+my %GENERALIZED_RESULTS = ((map { $_ => $_ } COMPLETE_RESULTS), (map { $_ => 'incomplete' } INCOMPLETE_RESULTS));
+
+sub generalize_result {
+    my ($concrete_result) = @_;
+    return $GENERALIZED_RESULTS{$concrete_result} // NONE;
+}
```

Comment:
I see. This looks like a similar approach I needed to take in https://github.com/os-autoinst/openQA/pull/2489/files#diff-85ae48e70a5c110c9e439c3a5ea28d5fR233-R246 . So it looks like first we had a manageable list of states+results, then we added more, now we have too many and we need to map them to less. I guess eventually we should look into the "states" and "results" again.